RoaringBitmap/src/main/java/org/roaringbitmap/longlong/HighLowContainer.java

```diff
@@ -111,6 +108,56 @@ public boolean isEmpty() {
     return art.isEmpty();
   }
 
+  private void assertNonEmpty() {
+    if(isEmpty()) {
+      throw new NoSuchElementException("Empty " + this.getClass().getSimpleName());
+    }
+  }
+
+  private LeafNode getExtremeNode(Node root, boolean reverse) {
+    Node parent = root;
```

Comment:
I think I'd prefer this function picking up the root node itself rather than being handed an arbitrary node and assuming it's the root. I'm concerned about a bunch of edge cases here where on arbitrary nodes `parent.getMaxPos()` (or min) may return `-1` (because `parent` is empty) and then `parent.getChild` will be unhappy. This shouldn't be possible to happen with a legal tree when starting from the root...although tbh I'm not quite sure right now what our guarantees about empty non-leaf nodes are when removing stuff from the tree. Maybe it's always illegal (probably should be^^).

Reply:
Should I return an exception if `int childIndex = reverse ? parent.getMaxPos() : parent.getMinPos();` is -1 ? I have no strong opinion either.

Reply:
I would hope `getChild` already does that? If not, please make it do that^^

Reply:
We already get ArrayIndexOutOfBounds in such a case (in `.getChild`)